### PR TITLE
Initial Submission Avneet 

### PR DESCRIPTION
**Pull request by avneetgill**

### CODE REVIEW

File: LogFileReader.cpp [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+
+    bool voltOk;
+    batteryData.voltage = sections.at(1).toDouble(&voltOk);
+    if(!voltOk)
+        return false;
+
+    bool curOk;
+    batteryData.current = sections.at(2).toDouble(&curOk);
+    if(!curOk)
```
Comment (Reddragoneer1) [inline LogFileReader.cpp]: We use strict bracketing conventions so one liners such as this should also have brackets

File: BatteryStateOfChargeService.cpp [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+namespace
+{
+    const double BATTERY_AMP_HOUR_CAPACITY = 123.0;
+}
+
+BatteryStateOfChargeService::BatteryStateOfChargeService(double initialStateOfChargePercent)
+: initialStateOfChargePercent_(initialStateOfChargePercent)
+{
+    ampHourUsed_ = 0.0;
```
Comment (bill-luu) [inline BatteryStateOfChargeService.cpp]: Use an [initialization list ](https://en.cppreference.com/w/cpp/language/initializer_list)for stuff in the constructor: 
```
BatteryStateOfChargeService::BatteryStateOfChargeService(double initialStateOfChargePercent)
: initialStateOfChargePercent_(initialStateOfChargePercent)
, timeTilCharged(QTime (0,0,0,0))
, timeTilDepleted(QTime (0,0,0,0))
, etc
```

File: BatteryStateOfChargeService.cpp [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+   else
+   {
+      return timeTilDepleted_;
+   }
+}
+
+void BatteryStateOfChargeService::addData(const BatteryData& batteryData)
+{
+    const double MILLISECONDS_TO_HOURS = 1/3600000.0;
```
Comment (bill-luu) [inline BatteryStateOfChargeService.cpp]: Constants should go in the namespace above.

File: BatteryStateOfChargeService.cpp [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    const double PERCENT_TO_DECIMAL = 0.01;
+        if(ampHourUsed_ == 0.0)
+        {
+             ampHourUsed_ = BATTERY_AMP_HOUR_CAPACITY*initialStateOfChargePercent_*PERCENT_TO_DECIMAL;
```
Comment (bill-luu) [inline BatteryStateOfChargeService.cpp]: keep spaces between operators: `BATTERY_AMP_HOUR_CAPACITY * initialStateOfChargePercent_ * PERCENT_TO_DECIMAL`

File: BatteryStateOfChargeService.cpp [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+        }
+        else
+        {
+             ampHourUsed_ += ((previousCurrent_+batteryData.current)/2)*((previousTime_).msecsTo(batteryData.time)*MILLISECONDS_TO_HOURS);
```
Comment (bill-luu) [inline BatteryStateOfChargeService.cpp]: This one line is doing a lot, splitting it into multiple calculations would improve readability.

File: BatteryStateOfChargeService.cpp [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+    {
+        hourCharge_ *= -1;
+    }
+    hourUsed_ = (BATTERY_AMP_HOUR_CAPACITY-ampHourUsed_)/batteryData.current;
+    QTime hourTilCharged = QTime (0,0,0,0);
+    QTime hourTilDepleted = QTime (0,0,0,0);
+    hourTilCharged = hourTilCharged.addMSecs((int)(hourCharge_/MILLISECONDS_TO_HOURS));
+    hourTilDepleted = hourTilDepleted.addMSecs((int)(hourUsed_/MILLISECONDS_TO_HOURS));
+    timeTilCharged_ = hourTilCharged;
+    timeTilDepleted_ = hourTilDepleted;
```
Comment (bill-luu) [inline BatteryStateOfChargeService.cpp]: It looks like you can go `timeTilCharged_ = QTime(0,0,0,0).addMSecs((int)(hourCharge_/MILLISECONDS_TO_HOURS));`?

File: LogFileReader.cpp [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+
+    if(sections.length() != 3)
+    {
+        return false;
+    }
+
+    QString timeString = sections.at(0);
+    batteryData.time = QTime::fromString(timeString, STRING_TIME_FORMAT);
+    bool time = batteryData.time.isValid();
+    if(time == false)
```
Comment (bill-luu) [inline LogFileReader.cpp]: `!time` instead of `if(time == false)`

File: LogFileReader.cpp [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+    QStringList sections = line.split(BATDATA_DELIMITER);
+
+    if(sections.length() != 3)
+    {
+        return false;
+    }
+
+    QString timeString = sections.at(0);
+    batteryData.time = QTime::fromString(timeString, STRING_TIME_FORMAT);
+    bool time = batteryData.time.isValid();
```
Comment (bill-luu) [inline LogFileReader.cpp]: This can be more descriptive, something like `timeOk`?

File: LogFileReader.cpp [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+    if(!voltageOk)
+    {
+        return false;
+    }
+
+    bool currentOk;
+    batteryData.current = sections.at(2).toDouble(&currentOk);
+    if(!currentOk)
+    {
+        return false;
```
Comment (bill-luu) [inline LogFileReader.cpp]: The last three checks can be combined into 1!

File: BatteryStateOfChargeService.cpp (hunk 2 above)
Comment (avneetgill) [reply BatteryStateOfChargeService.cpp]: Sounds good, I'll double check next time. Not sure how that happened!

**Pull request closed without merging**